### PR TITLE
🧪 Introduce @usedapp/testing

### PR DESCRIPTION
**Pull request by rzadp**
- Extract testing library from `core`, closes https://github.com/EthWorks/useDapp/issues/17
- Closes https://github.com/EthWorks/useDapp/issues/31
- Resolve https://github.com/EthWorks/useDapp/pull/28#discussion_r574630107
- Resolve https://github.com/EthWorks/useDapp/pull/28#discussion_r574629282
- Add possibility to add user wrapper, illustrated with `useAdder` example usage with hook arguments and provider wrapper

### CODE REVIEW

File: packages/core/src/constants/abi/index.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { Interface } from '@ethersproject/abi'
+import MultiCall from './MultiCall.json'
+
+const MultiCallABI = new Interface(MultiCall.abi)
```
Comment (marekkirejczyk) [inline packages/core/src/constants/abi/index.ts:4]: MultiCallABI -> MultiCallInterface

File: packages/core/package.json [hunk 2]
```diff
@@ -42,16 +34,13 @@
...
     "ethereum-waffle": "^3.2.2",
     "react": "^17.0.1",
-    "react-dom": "^17.0.1",
-    "styled-components": "^5.2.1"
+    "react-dom": "^17.0.1"
   },
   "scripts": {
-    "start": "webpack serve --mode development",
     "build": "tsc",
-    "test": "mocha -r jsdom-global/register",
```
Comment (marekkirejczyk) [inline packages/core/package.json:51]: No test in core?

File: packages/testing/examples/useBlockMeta.test.tsx [hunk 3]
```diff
@@ -1,27 +1,27 @@
 import { expect } from 'chai'
-import { useBlockMeta } from '../src'
-import { renderWeb3Hook, sleep } from '../src/testing'
+import { useBlockMeta } from '@usedapp/core'
+import { renderWeb3Hook, sleep } from '../src'
```
Comment (marekkirejczyk) [inline packages/testing/examples/useBlockMeta.test.tsx:3]: I wonder if it should be in 'examples' directory. It seems like a real test of core package.

File: packages/testing/package.json [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    "prettier": "^2.1.2",
+    "typescript": "^4.1.3"
+  },
+  "dependencies": {
+    "@ethersproject/providers": "^5.0.17",
+    "@types/react": "^17.0.0",
+    "@types/react-dom": "^17.0.0",
+    "@usedapp/core": "*",
+    "@web3-react/core": "^6.1.1",
+    "@web3-react/injected-connector": "^6.0.7",
```
Comment (marekkirejczyk) [inline packages/testing/package.json:35]: Is that dependency needed?

File: packages/testing/package.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "@usedapp/testing",
```
Comment (marekkirejczyk) [inline packages/testing/package.json:2]: @usedapp/testing or @usedapp/test? Which is better?

File: packages/testing/examples/useBlockNumber.test.tsx [hunk 6]
```diff
@@ -1,20 +1,20 @@
 import { expect } from 'chai'
-import { useBlockNumber } from '../src'
-import { renderWeb3Hook } from '../src/testing'
+import { useBlockNumber } from '@usedapp/core'
+import { renderWeb3Hook } from '../src'
 
 describe('useBlockNumber', () => {
   it('retrieves block number', async () => {
```
Comment (marekkirejczyk) [inline packages/testing/examples/useBlockNumber.test.tsx:6]: Should not be in 'example' folder?

**Review by marekkirejczyk — APPROVED** [review]
Looks good, some considerations in comments.

File: packages/testing/package.json (hunk 4 above)
Comment (rzadp) [reply packages/testing/package.json:35]: https://github.com/EthWorks/useDapp/pull/42

**Pull request merged**